gulpfile.js

```diff
@@ -38,7 +38,7 @@ gulp.task('ghpages', 'Deploy documentation to gh-pages', ['nojekyll'], function
 
 gulp.task("docs", 'Compile documentation from src code', function () {
   return gulp
-    .src(["src/**/*.ts"])
+    .src(["typings/globals/es6-promise/index.d.ts", "src/**/*.ts"])
```

Comment:
This is unrelated to changing filter types / interfaces though.  

Would rather fix typedoc in separate PR with all of the needed .d.ts files.  
I assume it needs all the dependent libraries, perhaps just include all typings folder.